docs/concepts/storage.md

```diff
@@ -0,0 +1,73 @@
+
+State-staked tokens are unavailable for other uses such as validation staking. This increases the yield that validators will receive. Learn more about this in [the economics whitepaper](https://near.org/papers/economics-in-sharded-blockchain/).
+
+## When do tokens get staked?
+
+On each incoming transaction that adds data.
+
+Let's walk through an example:
+
+1. You launch a [a custom token](https://examples.near.org/?q=fungible), deploying your token's smart contract to the account `mycoiniscoolerthanyours.near`
+2. Token holders can name an escrow – an account which can transfer tokens on their behalf. Naming an escrow will require a call to your smart contract, and your smart contract will need to store this mapping of token-owner-to-escrow-access.
+3. When this call comes in, NEAR will check that `mycoiniscoolerthanyours.near` has a large enough balance that it can stake an amount corresponding to this data addition. If it does not, the transaction will fail. Otherwise, these new tokens get locked for state staking and the transaction completes.
+
+Note in this example that it's the token holders and not your smart contract which, [by default](https://docs.near.org/docs/concepts/gas#what-about-prepaid-gas), pay for the gas fee.
+
+## The "million worthless escrows" attack
+
+Note that this can create an attack surface. To continue the example above, if naming an escrow costs a token holder close to nothing while costing the contract owner significantly more, then a malicious user can exploit the imbalance to make maintaining the contract prohibitively expensive.
+
+Take care, then, when designing your smart contracts to ensure that such attacks cost potential attackers more than it would be worth.
+
+## btw, you can remove data to unstake some tokens
+
+People familiar with the "immutable data" narrative about blockchains find this surprising. While it's true that an _indexing node_ will keep all data forever, _validating nodes_ (that is, the nodes run by most validators in the network) do not. Smart contracts can provide ways to delete data, and this data will be purged from most nodes in the network within a few [epochs](https://docs.near.org/docs/concepts/epoch).
+
+Note that a call to your smart contract to remove data has an associated gas fee. Also, since deletions takes time, there is an upper limit on the amount of data you can remove in a single transaction, as your transaction must be able to fit in the one-second block time.
+
+## How much does it cost?
+
+State staking is priced in an amount set by the network, which was initialized to **[9.09E19 yoctoNEAR per byte](https://github.com/nearprotocol/nearcore/blob/master/neard/res/genesis_config.json#L38)** and could be lowered (but not increased) in the future. NEAR's JSON RPC API provides [a way to query this initial setting](https://docs.near.org/docs/api/rpc-experimental#genesis-config), but does not yet provide a way to query the "live" configuration value.
```

Comment:
@bowenwang1996 is it true that `storage_amount_per_byte` will never be increased?

Reply:
No

Reply:
It could be increased? What would happen to all the contracts with insufficient free balance to stake for the higher cost?

Reply:
It depends on how we upgrade. For example we can make it such that the contract will be removed after some amount of time if no one funds it. Alternatively we can say that we will only allow transfer to this account until if has sufficient tokens to pay for storage.

Reply:
Still seems like something that could cause problems, even if mostly social problems. But thanks, I'll update the language.